### PR TITLE
EOS-16629 ISA-L: Replace Galois Arithmetic APIs with respective ISA APIs

### PR DESCRIPTION
**Pull request by sanjognaik**
- Used Intel ISA arithmetic APIs in m0_parity_mul() and m0_parity_div()
  functions.
- Intel ISA APIs are accessed in user space only.
- Galois APIs are called only in kernel space for now.
- Moved m0_parity_init() and m0_parity_fini() to kernel space as these
  are required to create and delete Galois tables only.

Signed-off-by: Sanjog Vikram Naik <sanjog.naik@seagate.com>

### CODE REVIEW

File: sns/parity_ops.c [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "lib/assert.h"
 #include "sns/parity_ops.h"
 
+#ifdef __KERNEL__
```
Comment (huanghua78) [inline sns/parity_ops.c:30]: I think we should keep the m0_parity_fini() and m0_parity_init() always defined for user space and kernel space.
```
M0_INTERNAL void m0_parity_fini(void)
{
#ifdef __KERNEL__
        galois_calc_tables_release();
#endif
}

M0_INTERNAL int m0_parity_init(void)
{
#ifdef __KERNEL__
        int ret = galois_create_mult_tables(M0_PARITY_GALOIS_W);
        M0_ASSERT(ret == 0);
#endif 
        return 0;
}
```

Comment (sanjognaik) [reply sns/parity_ops.c]: Done.

**Review by huanghua78 — APPROVED** [review]

File: Makefile.am [hunk 2]
```diff
@@ -185,7 +185,7 @@ motr_libmotr_la_CPPFLAGS  = -DM0_TARGET='libmotr' $(AM_CPPFLAGS)
 motr_libmotr_la_LDFLAGS   = -version-info @LT_VERSION@ -pthread $(AM_LDFLAGS)
 motr_libmotr_la_LIBADD    = @MATH_LIBS@ @PTHREAD_LIBS@ @AIO_LIBS@ @RT_LIBS@ \
                             @YAML_LIBS@ @PROFILER_LIBS@ @UUID_LIBS@ @GALOIS_LIBS@ \
```
Comment (huanghua78) [inline Makefile.am:187]: Here, just removing the "@GALOIS_LIBS@" will remove the linking with galois lib.

Comment (sanjognaik) [reply Makefile.am]: Done

**Comment by huanghua78** [comment]
Let's ask @shashank-parulekar to review this PR.

**Comment by huanghua78** [comment]
Jenkins building machines should be ready to test this:

https://jts.seagate.com/browse/EOS-16884

**Review by mukundkanekar — APPROVED** [review]

**Pull request merged**